src/consensus/raft/raft.h

```diff
@@ -507,12 +515,16 @@ namespace raft
 
       auto& node = nodes.at(to);
 
-      // Record the most recent index we have sent to this node.
-      node.sent_idx = end_idx;
-
       // The host will append log entries to this message when it is
       // sent to the destination node.
-      channels->send_authenticated(ccf::NodeMsgType::consensus_msg, to, ae);
+      if (!channels->send_authenticated(
```

Comment:
Small change here: as we can detect whether a message was successfully sent over the channel (i.e. `send_authenticated()` returns `false` if the channel is not yet established), let's not pretend we've sent the idx to the node if the channel is not yet established.